Assets/MixedRealityToolkit/_Core/Utilities/Scripts/Editor/Setup/EnforceEditorSettings.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.IO;
+using UnityEngine;
+using UnityEditor;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Utilities.Editor.Setup
+{
+    /// <summary>
+    /// Sets Force Text Serialization and visible meta files in all projects that use the Mixed Reality Toolkit.
+    /// </summary>
+    [InitializeOnLoad]
+    public class EnforceEditorSettings
+    {
+        private const string SessionKey = "_MixedRealityToolkit_Editor_ShownSettingsPrompts";
+
+        static EnforceEditorSettings()
+        {
+            if (!IsNewEditorSession())
+            {
+                return;
+            }
+
+            bool refresh = false;
+            bool restart = false;
+
+            if (EditorSettings.serializationMode != SerializationMode.ForceText)
+            {
+                if (EditorUtility.DisplayDialog(
+                        "Force Text Asset Serialization?",
+                        "The Mixed Reality Toolkit is easier to maintain if the asset serialization mode for this project is set to \"Force Text\". Would you like to make this change?",
+                        "Force Text Serialization",
+                        "Later"))
+                {
+                    EditorSettings.serializationMode = SerializationMode.ForceText;
+                    Debug.Log("Setting Force Text Serialization");
+                    refresh = true;
+                }
+            }
+
+            if (!EditorSettings.externalVersionControl.Equals("Visible Meta Files"))
+            {
+                if (EditorUtility.DisplayDialog(
+                    "Make Meta Files Visible?",
+                    "The Mixed Reality Toolkit would like to make meta files visible so they can be more easily handled with common version control systems. Would you like to make this change?",
+                    "Enable Visible Meta Files",
+                    "Later"))
+                {
+                    EditorSettings.externalVersionControl = "Visible Meta Files";
+                    Debug.Log("Updated external version control mode: " + EditorSettings.externalVersionControl);
+                    refresh = true;
+                }
+            }
+
+            if (PlayerSettings.scriptingRuntimeVersion != ScriptingRuntimeVersion.Latest)
```

Comment:
We have checks on the ScriptingRuntime and ScriptingConpatibility
Should we also not have one for the Scripting Backend, to ensure they use IL2CPP as the new standard for 2018?

Reply:
Done